src/parseTools.js

```diff
@@ -57,13 +57,17 @@ function preprocess(text, filenameHint) {
           const trimmed = line.trim();
           if (trimmed[0] === '#') {
             const first = trimmed.split(' ', 1)[0];
-            if (first == '#if' || first == '#ifdef') {
+            if (first == '#if' || first == '#ifdef' || first == '#elif') {
               if (first == '#ifdef') {
                 warn('warning: use of #ifdef in js library.  Use #if instead.');
               }
-              const after = trimmed.substring(trimmed.indexOf(' '));
-              const truthy = !!eval(after);
-              showStack.push(truthy);
+              if (first == '#elif' && showStack.pop()) {
+                showStack.push(false);
```

Comment:
Doesn't this have to evaluate `after`, or am I misunderstanding how `showStack` works? Based on the case for `#else`, I would expect something like this:

```
showStack.push(!showStack.pop() && !!(eval(after)))
```

Reply:
My understanding is that testing for `showStack.pop()` here is enough to tell us that the previous block/if was taken (it tells us we are exiting a block that was not being ignored), which means we don't need to evaluate `after` here at all... its doesn't matter what the condition is because we already processed one of the blocks in the if/elif chain.

However, you have made be realize that this doesn't handle changes with more than one `elif` where if any of the conditions evaluate to true then all the other can be ignored.

Reply:
I added a new state (and a test).